test-container/src/main/java/io/strimzi/StrimziKafkaCluster.java

```diff
@@ -0,0 +1,143 @@
+
+        additionalKafkaConfiguration.putAll(defaultKafkaConfigurationForMultiNode);
+
+        // multi-node set up
+        this.brokers = IntStream
+            .range(0, this.brokersNum)
+            .mapToObj(brokerId -> {
+                LOGGER.info("Starting broker with id {}", brokerId);
+                // adding broker id for each kafka container
+                additionalKafkaConfiguration.put("broker.id", String.valueOf(brokerId));
+
+                return new StrimziKafkaContainer(imageVersion, additionalKafkaConfiguration)
+                    .withNetwork(this.network)
+                    .withNetworkAliases("broker-" + brokerId)
+                    .dependsOn(this.zookeeper)
+                    .withExternalZookeeper("zookeeper:" + StrimziZookeeperContainer.ZOOKEEPER_PORT);
+            })
+            .collect(Collectors.toList());
+    }
+
+    public StrimziKafkaCluster(Map<String, String> additionalKafkaConfiguration) {
+        this(StrimziKafkaContainer.getStrimziVersion() + "-kafka-" + StrimziKafkaContainer.getLatestKafkaVersion(),
+            3,
+            3,
+            additionalKafkaConfiguration);
+    }
+
+    public Collection<StrimziKafkaContainer> getBrokers() {
+        return this.brokers;
+    }
+
+    public String getBootstrapServers() {
+        return brokers.stream()
+            .map(StrimziKafkaContainer::getBootstrapServers)
+            .collect(Collectors.joining(","));
+    }
+
+    @Override
+    public void start() {
+        Stream<Startable> startables = this.brokers.stream().map(Startable.class::cast);
+        try {
+            Startables.deepStart(startables).get(60, TimeUnit.SECONDS);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            e.printStackTrace();
+        }
+
+        TestUtils.waitFor("Zookeeper readiness", Duration.ofSeconds(1).toMillis(), Duration.ofSeconds(30).toMillis(),
+            () -> {
+                Container.ExecResult result = null;
+                try {
+                    result = this.zookeeper.execInContainer(
+                        "sh", "-c",
+                        "bin/zookeeper-shell.sh zookeeper:" + StrimziKafkaContainer.ZOOKEEPER_PORT + " ls /brokers/ids | tail -n 1"
+                    );
+                    String brokers = result.getStdout();
+
+                    LOGGER.info("Stdout from zookeeper container....{}", result.getStdout());
+
+                    return brokers != null && brokers.split(",").length == this.brokersNum;
+                } catch (IOException | InterruptedException e) {
+                    e.printStackTrace();
+                    return false;
+                }
+            });
+    }
+
+    @Override
+    public void stop() {
+        // stop all containers in parallel
+        Stream.concat(this.brokers.stream(), Stream.of(this.zookeeper))
```

Comment:
On the command line if I kill ZK first it sometimes prevents a broker from shutting down quickly. So maybe it's better to do the ZK shudown first, and then shut down the brokers in parallel.

Reply:
good point yes